### DISCUSSION TITLE
Mikro-ORM CLI from .env

### DISCUSSION THREAD

**Discussion by wguerram** [mikro-orm/mikro-orm#3069] [Q&A]
Hi, I'm using NX (Monorepo) and NestJS. Mikro-ORM configuration is being read from an .env file only. I've TS enable and I can start server without issues but in order to run migrations and seeders I have to enable this variable in .env

MIKRO_ORM_ENTITIES=./**/*.entity.js

The problem I have is that when that variable is set then I can't serve the app because Micro-ORM will try to read from folder the entities but that folder does not have any files because I'm using webpack. Also I don't understand why the CLI is requesting those values if they don't exists because of the webpack thing.

Is there a way to tell the CLI to ignore that variable MIKRO_ORM_ENTITIES and just use the .ts files?

`
MIKRO_ORM_TYPE=mariadb
MIKRO_ORM_HOST=localhost
MIKRO_ORM_PORT=3306
MIKRO_ORM_USER=
MIKRO_ORM_PASSWORD=
MIKRO_ORM_DB_NAME=
# MIKRO_ORM_ENTITIES=./**/*.entity.js # Enable to run seeders
MIKRO_ORM_ENTITIES_TS=src/**/*.entity.ts

MIKRO_ORM_CLI_USE_TS_NODE=true

MIKRO_ORM_MIGRATIONS_PATH_TS = ./src/migrations
MIKRO_ORM_MIGRATIONS_PATTERN=^[\w-]+\d+\.ts$ # ^[\w-]+\d+\.js$
MIKRO_ORM_MIGRATIONS_DROP_TABLES=false
MIKRO_ORM_MIGRATIONS_EMIT=ts

MIKRO_ORM_SEEDER_PATH = ./seeders
MIKRO_ORM_SEEDER_PATH_TS = ./src/database/seeders
MIKRO_ORM_SEEDER_DEFAULT_SEEDER = DatabaseSeeder
MIKRO_ORM_SEEDER_GLOB = '!(*.d).{js,ts}'
MIKRO_ORM_SEEDER_EMIT = true

MIKRO_ORM_POPULATE_AFTER_FLUSH = true
MIKRO_ORM_FORCE_ENTITY_CONSTRUCTOR = true
MIKRO_ORM_FORCE_UNDEFINED = true
`

**Comment by B4nan** [comment]
> Also I don't understand why the CLI is requesting those values if they don't exists because of the webpack thing.

You should _not_ use folder based discovery with webpack.

> Is there a way to tell the CLI to ignore that variable MIKRO_ORM_ENTITIES and just use the .ts files?

Nope, env vars always have precedence.

**Reply by wguerram** [reply]
Hi @B4nan, thanks for responding.

- I'm using env vars only
- I'm not using folder base discovery when running the application with webpack, NestJS allows me to import the entities and autoload them from the modules I'm using the entities. So the application itself is not doing folder base discovery. Actually I don't set a value on this var MIKRO_ORM_ENTITIES when running the application and works fine.
- My issue is the CLI, even though I'm declaring this env var MIKRO_ORM_CLI_USE_TS_NODE=true, the CLI is still expecting this var to be defined MIKRO_ORM_ENTITIES with any value. The problem when I set a value on MIKRO_ORM_ENTITIES my application is not running because it'll try to load entities from that folder.

So in order to run the application I have to comment MIKRO_ORM_ENTITIES in my env file and order to run the CLI I have to uncomment MIKRO_ORM_ENTITIES so that the value is set otherwise the CLI won't run. In the case of CLI I'm using folder base discovery with .ts files.

So how do I tell my application to not do folder discovery using env file but allow the CLI to do folder discovery with .ts files?

**Reply by B4nan** [reply]
> My issue is the CLI, even though I'm declaring this env var MIKRO_ORM_CLI_USE_TS_NODE=true, the CLI is still expecting this var to be defined MIKRO_ORM_ENTITIES with any value.

When you use CLI, there is no nestjs involved. So you need entity discovery to be configured, let it be folder based one (usable with env vars) or an array of entity references (which you cant do with env vars). Don't expect anything you configure in nestjs to have effect on the ORM CLI.

**Reply by B4nan** [reply]
You can always create a custom script that will use the CLI programatically. There you have absolute control, you can even remove some env vars from current environment before you call `MikroORM.init`.

**Reply by wguerram** [reply]
Thanks, I'll try the programatic approach with the CLI.
